src/Symfony/Component/Form/Extension/Core/DataTransformer/MoneyToLocalizedStringTransformer.php

```diff
@@ -59,7 +59,13 @@ public function transform($value)
                 throw new TransformationFailedException('Expected a numeric.');
             }
 
-            $value /= $this->divisor;
+            $value = (string) ($value / $this->divisor);
+
+            if (ctype_digit($value)) {
+                $value = (int) $value;
```

Comment:
you don't need the integer logic as it doesn't matter for `parent::transform` whether it's int or float. just cast it to float as it was a float before as well.